code/data_structures/src/avl_tree

```diff
@@ -0,0 +1,168 @@
+//The following is a task given as a part of screening test for internship
+//implemention of an AVL tree data structure
+#include<iostream>
+#include<algorithm>
+#define nullptr NULL
+
+
+struct Node 
+{
+    int data;
+    Node *left;
+    Node *right;
+    int height;
+}*root;
+class avl{
+	public:
+
+    Node *NewNode(int key) 
+{
+    Node *temp=( Node*)malloc(sizeof( Node));
+    temp->data=key;
+    temp->left=temp->right=nullptr;
+    return temp;
+}
+
+int height( Node *node) 
+{
+    if(node==nullptr) return 0;
+    int lh=height(node->left);
+    int rh=height(node->right);
+    if(lh > rh) return lh+1;
+    else return rh+1;
+}
+
+int getBalance(Node *node) 
+{
+    if(node==nullptr) return 0;
+    return height(node->left)-height(node->right);
+}
+
+Node *getRightRotate( Node *y) 
+{
+     Node *x=y->left;
+     Node *T2=x->right;
+    
+    x->right=y;
+    y->left=T2;
+    
+    y->height=std::max(height(y->left),height(y->right))+1;
+    x->height=std::max(height(x->left),height(x->right))+1;
+    
+    return x;
+    
+}
+
+Node *getLeftRotate( Node *x)
```

Comment:
Please use member functions.

Reply:
Its a member function belonging to  class avl.???